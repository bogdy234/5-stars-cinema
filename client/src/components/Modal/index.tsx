import { FC, ReactElement } from "react";

interface ModalProps {
  children: ReactElement;
  showModal: boolean;
}

const Modal: FC<ModalProps> = ({ children, showModal }): ReactElement => {
  return (
    <div
      className={`fixed z-10 left-0 top-0 w-full h-full overflow-auto bg-black bg-opacity-50 ${
        !showModal && "hidden"
      }`}
    >
      {children}
    </div>
  );
};

export default Modal;
