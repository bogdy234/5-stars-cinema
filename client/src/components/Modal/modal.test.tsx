import renderer from "react-test-renderer";
import Modal from "./index";

it("renders correctly", () => {
  const tree = renderer
    .create(<Modal children={<div>test</div>} showModal />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
