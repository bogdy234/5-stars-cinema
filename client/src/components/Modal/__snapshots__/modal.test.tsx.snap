// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="fixed z-10 left-0 top-0 w-full h-full overflow-auto bg-black bg-opacity-50 false"
>
  <div>
    test
  </div>
</div>
`;
